[Pipeline] Upgrade NuGet packages to match net10.0 target framework

## Problem

PR #182 upgraded the target framework to `net10.0` but left NuGet packages at .NET 8 versions. This causes 33 of 62 tests to fail with `InternalServerError` — the .NET 8 EF Core and ASP.NET Core packages are incompatible with the .NET 10 runtime.

Build succeeds (0 errors) but tests fail at runtime.

### Current package versions (wrong)

**TicketDeflection.csproj:**
- `Microsoft.EntityFrameworkCore.InMemory` 8.0.0

**TicketDeflection.Tests.csproj:**
- `Microsoft.AspNetCore.Mvc.Testing` 8.0.0
- `Microsoft.NET.Test.Sdk` 17.8.0
- `xunit` 2.5.3
- `xunit.runner.visualstudio` 2.5.3
- `coverlet.collector` 6.0.0

### Build warning
```
warning NU1903: Package 'Microsoft.Extensions.Caching.Memory' 8.0.0 has a known high severity vulnerability
```

## Acceptance Criteria

- [ ] Update `TicketDeflection/TicketDeflection.csproj`: `Microsoft.EntityFrameworkCore.InMemory` to latest 10.0.x (or 9.0.x if 10.0 not available yet)
- [ ] Update `TicketDeflection.Tests/TicketDeflection.Tests.csproj`: `Microsoft.AspNetCore.Mvc.Testing` to latest 10.0.x (or 9.0.x if 10.0 not available yet)
- [ ] Update `Microsoft.NET.Test.Sdk` to latest stable
- [ ] Update `xunit` and `xunit.runner.visualstudio` to latest stable
- [ ] `dotnet build TicketDeflection.sln` succeeds with 0 errors and 0 warnings
- [ ] `dotnet test TicketDeflection.sln` passes all 62 tests
- [ ] No `NU1903` vulnerability warnings remain

**IMPORTANT**: Do NOT modify any files in `.github/workflows/` — the PAT does not have workflow scope.

## Context

Post-Run-04 build fix. Framework upgraded to net10.0 in PR #182, but packages left at .NET 8 versions causing runtime test failures.

--- a/TicketDeflection.Tests/TicketDeflection.Tests.csproj
+++ b/TicketDeflection.Tests/TicketDeflection.Tests.csproj
@@ -10,11 +10,11 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="coverlet.collector" Version="6.0.0" />
-    <PackageReference Include="Microsoft.AspNetCore.Mvc.Testing" Version="8.0.0" />
-    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="17.8.0" />
-    <PackageReference Include="xunit" Version="2.5.3" />
-    <PackageReference Include="xunit.runner.visualstudio" Version="2.5.3" />
+    <PackageReference Include="coverlet.collector" Version="6.0.2" />
+    <PackageReference Include="Microsoft.AspNetCore.Mvc.Testing" Version="9.0.2" />
+    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="17.12.0" />
+    <PackageReference Include="xunit" Version="2.9.2" />
+    <PackageReference Include="xunit.runner.visualstudio" Version="2.8.2" />
   </ItemGroup>
 
   <ItemGroup>

--- a/TicketDeflection/TicketDeflection.csproj
+++ b/TicketDeflection/TicketDeflection.csproj
@@ -7,7 +7,7 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="Microsoft.EntityFrameworkCore.InMemory" Version="8.0.0" />
+    <PackageReference Include="Microsoft.EntityFrameworkCore.InMemory" Version="9.0.2" />
   </ItemGroup>
 
   <ItemGroup>

